### PR TITLE
ci: configure ci job to integrate with github-integration plugin for Jenkins

### PR DESCRIPTION
**Pull request by sameersbn**
We plan to switch to using the [github-integration](https://plugins.jenkins.io/github-pullrequest) 
plugin to handle the ci integration with github. This is mainly because the plugin allows us to 
configure behaviour of the pipeline based on github comments, labels and so on.

However there are some shortcomings of the plugin when configured for use with a multibranch jenkins 
pipeline, namely:

 - errors out while configuring the git clone operation
 - commit build status is not updated on github

This PR aims to workaround these two issues

### CODE REVIEW

**Comment by sameersbn** [comment]
Note: The PR is expected to fail until the changes are performed on the Jenkins server.

File: Jenkinsfile [hunk 1]
```diff
@@ -16,6 +16,51 @@ def parentZoneResourceGroup = 'jenkins-bkpr-rg'
...
+            desc = 'This commit cannot be built'
+            break
+        case 'pending':
+            desc = 'Waiting for status to be reported'
+            break
+        default:
+            return
+    }
+
+    withCredentials([usernamePassword(credentialsId: 'sameersbn-github', passwordVariable: 'GITHUB_TOKEN', usernameVariable: '')]) {
```
Comment (falfaro) [inline Jenkinsfile]: Are you using your own credentials?

Comment (sameersbn) [reply Jenkinsfile]: oh good catch. fixed!

**Comment by sameersbn** [comment]
bors retry

**Pull request merged**